addons/purchase_requisition/models/purchase_requisition.py

```diff
@@ -0,0 +1,303 @@
+    description = fields.Text(string='Description')
+    company_id = fields.Many2one('res.company', string='Company', required=True, default=lambda self: self.env.user._get_company())
+    purchase_ids = fields.One2many('purchase.order', 'requisition_id', string='Purchase Orders', states={'done': [('readonly', True)]})
+    po_line_ids = fields.One2many('purchase.order.line', compute='_compute_get_po_line', string='Products by vendor')
+    line_ids = fields.One2many('purchase.requisition.line', 'requisition_id', string='Products to Purchase', states={'done': [('readonly', True)]}, copy=True)
+    procurement_id = fields.Many2one('procurement.order', string='Procurement', ondelete='set null', copy=False)
+    warehouse_id = fields.Many2one('stock.warehouse', string='Warehouse')
+    state = fields.Selection([('draft', 'Draft'), ('in_progress', 'Confirmed'),
+                            ('open', 'Bid Selection'), ('done', 'PO Created'),
+                            ('cancel', 'Cancelled')], string='Status', track_visibility='onchange', required=True, copy=False, default='draft')
+    multiple_rfq_per_supplier = fields.Boolean(string='Multiple RFQ per vendor')
+    account_analytic_id = fields.Many2one('account.analytic.account', string='Analytic Account')
+    picking_type_id = fields.Many2one('stock.picking.type', string='Picking Type', required=True, default=_default_get_picking_in)
+
+    @api.depends('purchase_ids')
+    def _compute_get_po_line(self):
+        for element in self:
+            result = []
+            for po in element.purchase_ids:
+                result += [po_line.id for po_line in po.order_line]
+            element.po_line_ids = result
+
+    @api.multi
+    def tender_cancel(self):
+        # try to set all associated quotations to cancel state
+        for tender in self:
+            for purchase_order in tender.purchase_ids:
+                purchase_order.button_cancel()
+                purchase_order.message_post(body=_('Cancelled by the tender associated to this quotation.'))
+        return self.write({'state': 'cancel'})
+
+    @api.multi
+    def tender_in_progress(self):
+        if not all(obj.line_ids for obj in self):
+            raise UserError(_('You can not confirm call because there is no product line.'))
+        return self.write({'state': 'in_progress'})
+
+    @api.multi
+    def tender_open(self):
+        return self.write({'state': 'open'})
+
+    @api.multi
+    def tender_reset(self):
+        self.write({'state': 'draft'})
+        for p_id in self:
+            # Deleting the existing instance of workflow for PO
+            p_id.delete_workflow()
+            p_id.create_workflow()
```

Comment:
No need of iteration, as 'delete_workflow' and 'create_workflow' both the methods works on multiple records .